app/assets/javascripts/live_search.js

```diff
@@ -359,10 +361,25 @@
 
   LiveSearch.prototype.updateSortOptions = function updateSortOptions (results, action) {
     if (action !== this.serializeState(this.state)) { return }
-    this.updateElement(this.$sortBlock, results.sort_options_markup)
+    var currentSortOptions = this.$sortBlock.querySelector('select')
+    var newSortOptions
+
+    if (results.sort_options_markup) {
+      // The select element is removed from results.sort_options_markup in order to retain the original select element. Only the option
+      // elements are updated with new markup. This is to prevent focus from being lost on the select element after the user sorts their
+      // results.
+      newSortOptions = this.removeSelectElement(results.sort_options_markup)
+    }
+
+    this.updateElement(currentSortOptions, newSortOptions)
```

Comment:
Shouldn't all this code be inside the if statement on line 367? As all this code seems to only be relevant if `results.sort_options_markup` exists?